sdk/src/dataMapper/PortalDataMapper.ts

```diff
@@ -203,111 +237,38 @@ export default class PortalDataMapper extends BaseDataMapper<Portal, Portal_filt
   }
 
-  async simulateRegister(id: Address, name: string, description: string, isRevocable: boolean, ownerName: string) {
-    return this.simulatePortalRegistryContract("register", [id, name, description, isRevocable, ownerName]);
-  }
-
-  async register(
-    id: Address,
-    name: string,
-    description: string,
-    isRevocable: boolean,
-    ownerName: string,
-    waitForConfirmation: boolean = false,
-  ) {
-    const request = await this.simulateRegister(id, name, description, isRevocable, ownerName);
-    return executeTransaction(request, this.web3Client, this.walletClient, waitForConfirmation);
-  }
-
-  async simulateDeployDefaultPortal(
-    modules: Address[],
-    name: string,
-    description: string,
-    isRevocable: boolean,
-    ownerName: string,
-  ) {
-    return this.simulatePortalRegistryContract("deployDefaultPortal", [
-      modules,
-      name,
-      description,
-      isRevocable,
-      ownerName,
-    ]);
-  }
-
-  async deployDefaultPortal(
-    modules: Address[],
-    name: string,
-    description: string,
-    isRevocable: boolean,
-    ownerName: string,
-    waitForConfirmation: boolean = false,
-  ) {
-    const request = await this.simulateDeployDefaultPortal(modules, name, description, isRevocable, ownerName);
-    return executeTransaction(request, this.web3Client, this.walletClient, waitForConfirmation);
-  }
-
-  async getPortalByAddress(id: Address) {
-    return this.executePortalRegistryReadMethod("getPortalByAddress", [id]);
-  }
-
-  async isPortalRegistered(id: Address) {
-    return this.executePortalRegistryReadMethod("isRegistered", [id]);
-  }
-
-  private async executePortalRegistryReadMethod(functionName: string, args: unknown[]) {
-    return this.web3Client.readContract({
-      abi: abiPortalRegistry,
-      address: this.conf.portalRegistryAddress,
-      functionName,
-      args,
-    });
-  }
-
-  private async simulatePortalRegistryContract(functionName: string, args: unknown[]) {
-    if (!this.walletClient) throw new Error("VeraxSDK - Wallet not available");
-    try {
-      const { request } = await this.web3Client.simulateContract({
-        address: this.conf.portalRegistryAddress,
-        abi: abiPortalRegistry,
-        functionName,
-        account: this.walletClient.account,
-        args,
-      });
-
-      return request;
-    } catch (err) {
-      handleError(ActionType.Simulation, err);
-    }
-  }
-
   private async simulatePortalContract(
     portalAddress: Address,
     functionName: string,
     args: unknown[],
     value: bigint = 0n,
+    customAbi?: Abi,
   ) {
     if (!this.walletClient) throw new Error("VeraxSDK - Wallet not available");
+
+    const abi = [...abiDefaultPortal, ...(customAbi || [])];
+
     try {
       const { request } = await this.web3Client.simulateContract({
         address: portalAddress,
-        abi: abiDefaultPortal,
+        abi, // Simplified to directly use `abi`
```

Comment:
No need for this comment
```suggestion
        abi,
```

Reply:
done